### PR TITLE
[Quantum] Add "Credits for all" support to the CLI

### PR DESCRIPTION
**Pull request by warren-jones**
Automatically add providers participating in the "Credits for all" promotion when a workspace is created.

Because a new "autoAdd" flag was added in the Quantum Service API, the CLI's control-plane client requires an update.  A script that generates either the data-plane client or the control-plane client is included in this PR, along with the new AutoRest-generated client files that support API version 2022-01-10-preview.

Note: The files in src\quantum\azext_quantum\vendored_sdks were generated by the Microsoft AutoRest Code Generator according to parameters in src\quantum\eng\Generate-Client.ps1

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related commands
az quantum workspace create
az quantum offerings list

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

### CODE REVIEW

**Comment by yonzhan** [comment]
Quantum

**Comment by wangzelin007** [comment]
Hi [warren-jones](https://github.com/warren-jones),
Do these features need to be released before Microsoft Build 2022 (2022-05-24)
If not, can we move this pr to next milestone Jun 2022 (2022-07-05) ?

File: src/quantum/eng/Generate-Client.ps1 [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# --------------------------------------------------------------------------------------------
```
Comment (ricardo-espinoza) [inline src/quantum/eng/Generate-Client.ps1:1]: Comment only: I'm glad that this got added to the repo. While it's not a part of this PR, I'd recommend a follow up change adding a Readme.md in this directory explaining how to use it as well as the other script.

Comment (ricardo-espinoza) [inline src/quantum/eng/Generate-Client.ps1:1]: Is this file a replacement for Generate-DataPlane-Client.ps1, or will we only use this to generate the control plane one?

**Review by ricardo-espinoza — APPROVED** [review]

**Comment by ricardo-espinoza** [comment]
LGTM, recommend changing it to 'ready for review'.

File: src/quantum/eng/Generate-Client.ps1 (hunk 1 above)
Comment (warren-jones) [reply src/quantum/eng/Generate-Client.ps1:1]: Yes, a Readme.md is a good idea.  I planned to write a wiki page about this procedure, but a Readme.md would be easier to find.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**